packages/@react-spectrum/color/test/HexColorField.test.js

```diff
@@ -0,0 +1,295 @@
+    let {getByLabelText} = renderComponent({validationState: 'valid'});
+    let hexColorField = getByLabelText('Primary Color');
+    expect(hexColorField).not.toBeInvalid();
+  });
+
+  it('should show invalid validation state', function () {
+    let {getByLabelText} = renderComponent({validationState: 'invalid'});
+    let hexColorField = getByLabelText('Primary Color');
+    expect(hexColorField).toBeInvalid();
+  });
+
+  it('should be disabled', function () {
+    let {getByLabelText} = renderComponent({isDisabled: true});
+    let hexColorField = getByLabelText('Primary Color');
+    expect(hexColorField).toBeDisabled();
+  });
+
+  it('should be readonly', function () {
+    let {getByLabelText} = renderComponent({isReadOnly: true});
+    let hexColorField = getByLabelText('Primary Color');
+    expect(hexColorField).toHaveAttribute('readonly');
+  });
+
+  it('should be required', function () {
+    let {getByLabelText} = renderComponent({isRequired: true});
+    let hexColorField = getByLabelText(/Primary Color/);
+    expect(hexColorField).toBeRequired();
+  });
+
+  it('should be empty when invalid value is provided', function () {
+    let {getByLabelText} = renderComponent({defaultValue: true});
+    let hexColorField = getByLabelText('Primary Color');
+    expect(hexColorField.value).toBe('');
+
+    act(() => {hexColorField.focus();});
+    act(() => {hexColorField.blur();});
+    expect(hexColorField.value).toBe('');
+  });
+
+  it.each`
+    Name                                 | props
+    ${'3-length hex string'}             | ${{defaultValue: '#abc'}}
+    ${'6-length hex string'}             | ${{defaultValue: '#aabbcc'}}
+    ${'Color object'}                    | ${{defaultValue: new Color('#abc')}}
+    ${'3-length hex string controlled'}  | ${{value: '#abc'}}
+    ${'6-length hex string controlled'}  | ${{value: '#aabbcc'}}
+    ${'Color object controlled'}         | ${{value: new Color('#abc')}}
+  `('should accept $Name as value', function ({props}) {
+    let {getByLabelText} = renderComponent(props);
+    let hexColorField = getByLabelText('Primary Color');
+    expect(hexColorField.value).toBe('#AABBCC');
+  });
+
+  it.each`
+    Name                   | props
+    ${'custom min value'}  | ${{defaultValue: '#aaa', minValue: '#bbb'}}
+    ${'custom max value'}  | ${{defaultValue: '#ccc', maxValue: '#bbb'}}
+  `('should clamp initial value provided to $Name', function ({props}) {
+    let {getByLabelText} = renderComponent(props);
+    let hexColorField = getByLabelText('Primary Color');
+    expect(hexColorField.value).toBe('#BBBBBB');
+  });
+
+  it('should handle uncontrolled state', function () {
+    let onChangeSpy = jest.fn();
+    let {getByLabelText} = renderComponent({defaultValue: '#abc', onChange: onChangeSpy});
+
+    let hexColorField = getByLabelText('Primary Color');
+    expect(hexColorField.value).toBe('#AABBCC');
+
+    act(() => {hexColorField.focus();});
+    userEvent.clear(hexColorField);
```

Comment:
when a field is cleared, I think we should expect onChange to be called, maybe we should have an NaN equivalent for colors or return undefined?
Otherwise if a field starts empty, a user types a color, then deletes said color, it'll never go back to being a blank empty field

Reply:
Progress
- [ ] wrap `userEvent` in `act`
- [ ] change `onChange` behavior for when user clears the input.